tools/converter/synapse2ballerina/src/main/java/org/wso2/ei/tools/synapse2ballerina/visitor/SynapseConfigVisitor.java

```diff
@@ -302,10 +307,91 @@ public void visit(CallMediator mediator) {
         if (logger.isDebugEnabled()) {
             logger.debug("CallMediator");
         }
-        BallerinaProgramHelper.addImport(ballerinaASTModelBuilder, Constant.BLANG_HTTP, importTracker);
-        //TODO: Refactor this part to support other types of endpoints as well
-        IndirectEndpoint indirectEndpoint = (IndirectEndpoint) mediator.getEndpoint();
+        this.visit(mediator.getEndpoint());
+    }
+
+    /**
+     * @param endpoint
+     */
+    public void visit(Endpoint endpoint) {
+        if (endpoint instanceof IndirectEndpoint) {
```

Comment:
This will get triggered by CallMediator's visit method and we don't know what type of endpoint that it contains. That's why the instanceof check is needed.  We can move the instance of check to callmeditor's visit method, but then again if any other mediator needs to call an endpoint perhaps send mediator,  then we might be duplicating code.